pages/en/lb3/Polymorphic-relations.md

```diff
@@ -96,47 +142,143 @@ And:
 
-For example, "ImageLink":
+{% include code-caption.html content="common/models/picture.json" %}
 
-{% include code-caption.html content="/common/models/ImageLink.json" %}
 ```javascript
 {
-  "name": "ImageLink",
+  "name": "Picture",
   "base": "PersistedModel",
   ...
   "relations": {
-    "author": {
+    "imageable": {
+      "type": "belongsTo",
+      "polymorphic": true
+    }
+  },
+...
+
+// Alternatively, use an object for setup
+
+{
+  "name": "Picture",
+  "base": "PersistedModel",
+  ...
+  "relations": {
+    "imageable": {
       "type": "belongsTo",
-      "model": "Author",
-      "foreignKey": "authorId"
-    },
-    "pictures": {
-      "type": "hasMany",
-      "model": "Picture",
       "polymorphic": {
         "foreignKey": "imageableId",
         "discriminator": "imageableType"
       }
     }
   },
 ...
-}
 ```
 
+Or, in code:
+
+{% include code-caption.html content="common/models/picture.js" %}
+```javascript
+Picture.belongsTo('imageable', {
+  polymorphic: true
+}); 
+// Alternatively, use an object for setup
+Picture.belongsTo('imageable', {
+  polymorphic: {
+    foreignKey: 'imageableId',
+    discriminator: 'imageableType'
+  }
+});
+```
+
+## HasOne polymorphic relations
+
+The difference between hasOne and hasMany is, hasOne represents a "one-to-one" relation while hasMany represents a "one-to-many" relation.
+(I am going to add hyperlink of hasOne and hasMany here, haven't figured out how to link to another page :P)
+
+The relation definitions in 'HasOne' is almost same as 'HasMany', the only difference is `type` should be 'hasOne', for details, refer to [HasMany polymorphic relations](HasMany-polymorphic-relations).
+
+The following code shows how to dynamically define a 'HasOne' polymorphic relation, and also redefines the relation name with `as`: you can specify `as: 'avatar'` to explicitly set the name of the relation. If not set, it defaults to the polymorphic relation name.
+
+{% include code-caption.html content="/common/models/model.js" %}
+```javascript
+Picture.belongsTo('imageable', {polymorphic: true});
+Author.hasOne(Picture, {as: 'avatar', polymorphic: 'imageable'});
+Reader.hasOne(Picture, {polymorphic: {as: 'imageable'}});
+
+// To create a picture belongs to an author, you can use the method below
+author.avatar.create();
+```
+
+## HasManyThrough polymorphic relations
+
+To define a hasManyThrough polymorphic relation, there must be a "through" model, for example:
+
+> Author hasMany Picture through ImageLink polymorphically
+>
+> Reader hasMany Picture through ImageLink polymorphically
+
+A hasManyThrough polymorphic relation means the through model has three properties created. The first two are same as those created in the toModel in a hasMany polymorphic relation: one is discriminator(the value is either `Author` or `Reader`) and the other one is foreignKey(the id(primarykey) value of either an author or a reader). The third property is a foreignKey property reference model Picture.
```

Comment:
It'd help users understand the last sentence more if we specify the type of model it references i.e. `The third property is a foreignKey property which references the modelFrom(?), in this case the Picture model.`